pong.cpp

```diff
@@ -0,0 +1,29 @@
+#include "pong.h"
+
+#include "ball.h"
+#include "opponent.h"
+#include "player.h"
+
+void Pong::PongInit()
+{
+    auto player = std::make_unique<Player>();
+    player->OnStart();
+    mGameObjects.push_back(std::move(player));
+
+    auto opponent = std::make_unique<Opponent>();
+    opponent->OnStart();
+    mGameObjects.push_back(std::move(opponent));
+
+    auto ball = std::make_unique<Ball>();
+    ball->OnStart();
+    mGameObjects.push_back(std::move(ball));
+}
+
+void Pong::PongGameLoop()
+{
+    for (auto& gameObject : mGameObjects)
```

Comment:
Inspired shit 👏 

See how much nicer that is? You make it impossible to have an out of bounds error. and if the type changes you have it `auto`matically updated here. Good stuff

Reply:
Yeah definitely. Theres really something to all this standard library and modern c++ stuff. Where has it been all my life 🤔